akka-stream/src/main/scala/akka/stream/impl/Stages.scala

```diff
@@ -147,6 +151,28 @@ import akka.stream.Attributes._
 
     val inputBoundary = name("input-boundary")
     val outputBoundary = name("output-boundary")
+
+    private val cache = new ConcurrentHashMap[(String, Class[_]), String]
+    private val verboseOperatorNames =
+      // unfortunately this must be a static setting determined from the classpath instead of from the ActorSystem
+      // as there is no ActorSystem necessary to create operators.
+      ConfigFactory.load().getBoolean("akka.stream.verbose-operator-names")
+
+    def nameForLambda(prefix: String, lambda: AnyRef): String =
+      if (verboseOperatorNames)
+        cache.computeIfAbsent(
+          (prefix, lambda.getClass), {
+            case (prefix, clazz) =>
+              def locationName: String = LineNumbers(lambda) match {
```

Comment:
LineNumbers requires an actual object around instead of just the class (necessary to extract `SerializedLambda` which contains the origin information), luckily we have one around.